packages/@eventual/core-runtime/src/local/clients/queue-client.ts

```diff
@@ -128,18 +158,64 @@ export class LocalQueueClient extends QueueClient {
   }
 }
 
+interface SerializedMessage
+  extends Omit<
+    FifoQueueHandlerMessageItem | StandardQueueHandlerMessageItem,
+    "sent"
+  > {
+  sent: number;
+  visibility?: number;
+}
+
 export class LocalQueue {
   constructor(
     private queue: FifoQueue | Queue,
-    private localConnector: LocalEnvConnector
+    private localConnector: LocalEnvConnector,
+    private messages: (
+      | FifoQueueHandlerMessageItem
+      | StandardQueueHandlerMessageItem
+    )[] = [],
+    private messageVisibility: Record<string, Date> = {}
   ) {}
 
-  public messages: (
-    | FifoQueueHandlerMessageItem
-    | StandardQueueHandlerMessageItem
-  )[] = [];
+  public serialize(): Buffer {
+    return Buffer.from(
+      JSON.stringify(
+        this.messages.map(({ sent, ...m }) => {
+          return {
+            ...m,
+            visibility: this.messageVisibility[m.id]?.getTime(),
+            sent: sent.getTime(),
+          };
+        }) satisfies SerializedMessage[]
+      )
+    );
+  }
 
-  public messageVisibility: Record<string, Date> = {};
+  public static fromSerializedData(
+    queue: FifoQueue | Queue,
+    localConnector: LocalEnvConnector,
+    data?: Buffer
+  ) {
+    if (!data) {
+      return new LocalQueue(queue, localConnector);
+    }
+    const messageVisibility: Record<string, Date> = {};
+    const storedMessages = JSON.parse(
+      data.toString("utf-8")
+    ) as SerializedMessage[];
+    const messages = storedMessages.map(({ sent, visibility, ...m }) => {
+      const message = {
+        ...m,
+        sent: new Date(sent),
+      };
+      if (visibility) {
+        messageVisibility[m.id] = new Date(visibility);
+      }
+      return message;
```

Comment:
Are there any timers that need to be restored?

Reply:
they are in the `time` files, aka timecontroller